### PR TITLE
feat(Sink): support ck Replicated*tree

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/replication
Replicated*tree for ck is supported, while the previous batch of sending once per chunk was too small, so here it is modified to send by barriers

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/connector/src/sink/clickhouse.rs [hunk 1]
```diff
@@ -25,7 +26,7 @@ use risingwave_common::types::{DataType, Decimal, ScalarRefImpl, Serial};
 use serde::ser::{SerializeSeq, SerializeStruct};
 use serde::Serialize;
 use serde_derive::Deserialize;
-use serde_with::serde_as;
+use serde_with::{serde_as, DisplayFromStr};
```
Comment (hzxa21) [inline src/connector/src/sink/clickhouse.rs:29]: This change is not needed.

File: src/connector/src/sink/clickhouse.rs [hunk 2]
```diff
@@ -122,6 +144,43 @@ impl ClickHouseEngine {
...
+                Ok(ClickHouseEngine::ReplicatedAggregatingMergeTree)
+            }
+            // ReplicatedVersionedCollapsingMergeTree("a","b",sign_name,"c")
+            "ReplicatedVersionedCollapsingMergeTree" => {
+                let sign_name = engine_name
+                    .create_table_query
+                    .split("ReplicatedVersionedCollapsingMergeTree(")
+                    .last()
+                    .ok_or_else(|| SinkError::ClickHouse("must have last".to_string()))?
+                    .split(',')
```
Comment (hzxa21) [inline src/connector/src/sink/clickhouse.rs:160]: do we need to `trim` spaces after the spilt? same for other places.

File: src/connector/src/sink/clickhouse.rs [hunk 3]
```diff
@@ -534,6 +600,15 @@ impl AsyncTruncateSinkWriter for ClickHouseSinkWriter {
     ) -> Result<()> {
         self.write(chunk).await
     }
+
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<()> {
+        if let Some(inserter) = self.inserter.take()
```
Comment (hzxa21) [inline src/connector/src/sink/clickhouse.rs:605]: Should we expose a config for user to determine the batch size of each insert?

**Review by hzxa21 — APPROVED** [review]
Generally LGTM

File: src/connector/src/sink/clickhouse.rs (hunk 3 above)
Comment (xxhZs) [reply src/connector/src/sink/clickhouse.rs:605]: It doesn't seem to work, the ck rust interface doesn't allow it to be configured

Comment (hzxa21) [reply src/connector/src/sink/clickhouse.rs:605]: I mean does it make sense to end (flush) an inserted based on batch size?

**Pull request merged**